File: word.py
import requests
from bs4 import BeautifulSoup

def scan_wordpress(url):
    try:
        # Mengirim permintaan GET ke URL
        response = requests.get(url)
        response.raise_for_status()  # Memastikan permintaan berhasil

        # Memeriksa versi WordPress
        if 'generator' in response.text:
            soup = BeautifulSoup(response.text, 'html.parser')
            generator = soup.find('meta', attrs={'name': 'generator'})
            if generator:
                print(f"WordPress Version: {generator['content']}")
            else:
                print("WordPress version not found.")
        else:
            print("This site does not appear to be a WordPress site.")

        # Mencari tema yang digunakan
        theme_url = url + '/wp-content/themes/'
        themes_response = requests.get(theme_url)
        if themes_response.status_code == 200:
            print("Themes Directory is accessible.")
            themes = themes_response.text.splitlines()
            for line in themes:
                if 'href' in line:
                    print(f"Theme found: {line.split('href=')[1].split('>')[0]}")
        else:
            print("Themes Directory is not accessible.")

        # Mencari plugin yang digunakan
        plugins_url = url + '/wp-content/plugins/'
        plugins_response = requests.get(plugins_url)
        if plugins_response.status_code == 200:
            print("Plugins Directory is accessible.")
            plugins = plugins_response.text.splitlines()
            for line in plugins:
                if 'href' in line:
                    print(f"Plugin found: {line.split('href=')[1].split('>')[0]}")
        else:
            print("Plugins Directory is not accessible.")

    except requests.exceptions.RequestException as e:
        print(f"An error occurred: {e}")

# Ganti dengan URL situs WordPress yang ingin Anda scan
url = 'http://example.com'
scan_wordpress(url)

File: word.php
<?php
function scan_wordpress($url) {
    // Mengatur opsi untuk permintaan HTTP
    $options = [
        "http" => [
            "header" => "User -Agent: PHP\r\n"
        ]
    ];
    $context = stream_context_create($options);

    // Mengambil konten dari URL
    $response = @file_get_contents($url, false, $context);
    
    if ($response === FALSE) {
        echo "Error: Unable to access the URL.\n";
        return;
    }

    // Memeriksa versi WordPress
    if (preg_match('/<meta name="generator" content="WordPress (.*?)"/', $response, $matches)) {
        echo "WordPress Version: " . htmlspecialchars($matches[1]) . "<br>";
    } else {
        echo "This site does not appear to be a WordPress site.<br>";
        return;
    }

    // Mencari tema yang digunakan
    $theme_url = $url . '/wp-content/themes/';
    $themes_response = @file_get_contents($theme_url, false, $context);
    if ($themes_response !== FALSE) {
        echo "Themes Directory is accessible.<br>";
        preg_match_all('/href="([^"]+)"/', $themes_response, $theme_matches);
        foreach ($theme_matches[1] as $theme) {
            if (strpos($theme, '/wp-content/themes/') !== false) {
                echo "Theme found: " . htmlspecialchars(basename($theme)) . "<br>";
            }
        }
    } else {
        echo "Themes Directory is not accessible.<br>";
    }

    // Mencari plugin yang digunakan
    $plugins_url = $url . '/wp-content/plugins/';
    $plugins_response = @file_get_contents($plugins_url, false, $context);
    if ($plugins_response !== FALSE) {
        echo "Plugins Directory is accessible.<br>";
        preg_match_all('/href="([^"]+)"/', $plugins_response, $plugin_matches);
        foreach ($plugin_matches[1] as $plugin) {
            if (strpos($plugin, '/wp-content/plugins/') !== false) {
                echo "Plugin found: " . htmlspecialchars(basename($plugin)) . "<br>";
            }
        }
    } else {
        echo "Plugins Directory is not accessible.<br>";
    }
}

// Memeriksa apakah form telah disubmit
if ($_SERVER["REQUEST_METHOD"] == "POST" && !empty($_POST['url'])) {
    $url = trim($_POST['url']);
    // Menambahkan http:// jika tidak ada
    if (!preg_match("~^(?:f|ht)tps?://~i", $url)) {
        $url = "http://" . $url;
    }
    scan_wordpress($url);
}
?>

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>WordPress Scanner</title>
</head>
<body>
    <h1>WordPress Scanner</h1>
    <form method="post" action="">
        <label for="url">Enter WordPress Site URL:</label>
        <input type="text" id="url" name="url" required>
        <input type="submit" value="Scan">
    </form>
</body>
</html>
